Indicators/RelativeVigorIndex.cs

```diff
@@ -0,0 +1,210 @@
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RelativeVigorIndex"/> (RVI) class along with a signal term.
+        /// </summary>
+        /// <param name="name">The name of this indicator.</param>
+        /// <param name="signalName">The name of the signal associated with this indicator.</param>
+        /// <param name="period">The period for the RelativeVigorIndex.</param>
+        /// <param name="type">The type of Moving Average to use</param>
+        public RelativeVigorIndex(
+            string name, string signalName, int period, MovingAverageType type = MovingAverageType.Simple)
+            : base(name)
+        {
+            WarmUpPeriod = period + 3;
+            CloseBand = type.AsIndicator("_closingBand", period);
+            RangeBand = type.AsIndicator("_rangeBand", period);
+            _previousInputs = new RollingWindow<IBaseDataBar>(3);
+            Signal = new RviSignal(signalName);
+        }
+
+        /// <summary>
+        /// Computes the next value of this indicator from the given state
+        /// </summary>
+        /// <param name="input">The input given to the indicator</param>
+        /// <returns>A new value for this indicator</returns>
+        protected override decimal ComputeNextValue(IBaseDataBar input)
+        {
+            if (_previousInputs.IsReady)
+            {
+                var a = input.Close - input.Open;
+                var b = _previousInputs[0].Close - _previousInputs[0].Open;
+                var c = _previousInputs[1].Close - _previousInputs[1].Open;
+                var d = _previousInputs[2].Close - _previousInputs[2].Open;
+                var e = input.High - input.Low;
+                var f = _previousInputs[0].High - _previousInputs[0].Low;
+                var g = _previousInputs[1].High - _previousInputs[1].Low;
+                var h = _previousInputs[2].High - _previousInputs[2].Low;
+                CloseBand.Update(input.Time, (a + 2 * (b + c) + d) / 6);
+                RangeBand.Update(input.Time, (e + 2 * (f + g) + h) / 6);
+            }
+
+            _previousInputs.Add(input);
+            if (CloseBand.IsReady && RangeBand.IsReady)
+            {
+                var rvi = CloseBand / RangeBand;
+                Signal?.Update(input.Time, rvi); // Checks for null before updating.
+                return rvi;
+            }
+
+            return 0m;
+        }
+
+        /// <summary>
+        /// Resets this indicator to its initial state
+        /// </summary>
+        public override void Reset()
+        {
+            base.Reset();
+            CloseBand.Reset();
+            RangeBand.Reset();
+            _previousInputs.Reset();
+            Signal?.Reset();
+        }
+
+        /// <summary>
+        /// The signal for the Relative Vigor Index, itself an indicator. 
+        /// </summary>
+        public class RviSignal : Indicator, IIndicatorWarmUpPeriodProvider
```

Comment:
Cool! Let's please move this guy to his own file with his own simple test collection too 👍

Reply:
I ended up moving it to its own file but in a manner that it should only be accessible through the (main) RVI class. As such, I added `TestDivByZero` for it (to ensure it can properly avoid dividing by zero) in the test file for RVI.